packages/aws-cdk/lib/api/nested-stack-helpers.ts

```diff
@@ -0,0 +1,143 @@
+import * as path from 'path';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as fs from 'fs-extra';
+import { debug } from '../logging';
+import { ISDK } from './aws-auth';
+import { LazyListStackResources, ListStackResources } from './evaluate-cloudformation-template';
+import { CloudFormationStack, Template } from './util/cloudformation';
+
+/**
+ * Reads the currently deployed template from CloudFormation and adds a
+ * property, `NestedTemplate`, to any nested stacks that appear in either
+ * the deployed template or the just synthesized template. `NestedTemplate`
+ * is populated with contents of the nested template. This is done for all
+ * nested stack resources to arbitrary depths.
+ */
+export async function readCurrentTemplateWithNestedStacks(
+  rootStackArtifact: cxapi.CloudFormationStackArtifact, sdk: ISDK,
+): Promise<TemplateWithStackNames> {
+  const deployedTemplate = await readCurrentTemplate(rootStackArtifact, sdk);
+  const nestedStackNames = {};
+  await addNestedTemplatesToGeneratedAndDeployedStacks(rootStackArtifact, sdk, {
+    generatedTemplate: rootStackArtifact.template,
+    deployedTemplate: deployedTemplate,
+    deployedStackName: rootStackArtifact.stackName,
+  }, nestedStackNames);
+  return {
+    deployedTemplate,
+    nestedStackNames,
+  };
+}
+
+/**
+ * Returns the currently deployed template from CloudFormation that corresponds to `stackArtifact`.
+ */
+export async function readCurrentTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, sdk: ISDK): Promise<Template> {
```

Comment:
What do you think about changing the names of the two public functions from this file, so they're not confused with the currently identically named methods in `CloudFormationDeployments`?

Something simple like `loadCurrentTemplateWithNestedStacks()` / `loadCurrentTemplate()`?